src/components/input.js

```diff
@@ -41,21 +41,54 @@ const styles = {
   }
 }
 
-export const textInput = function(props) {
-  return h(Interactive, _.merge(
-    {
-      as: 'input',
-      className: 'focus-style',
-      style: {
-        ...styles.input,
-        width: '100%',
-        paddingLeft: '1rem', paddingRight: '1rem',
-        fontWeight: 400, fontSize: 14,
-        backgroundColor: props.disabled ? colors.grayBlue[4] : undefined
+export const textInput = props => h(Interactive,
+  _.merge({
+    as: 'input',
+    className: 'focus-style',
+    style: {
+      ...styles.input,
+      width: '100%',
+      paddingLeft: '1rem', paddingRight: '1rem',
+      fontWeight: 400, fontSize: 14,
+      backgroundColor: props.disabled ? colors.grayBlue[4] : undefined
+    }
+  },
+  props)
+)
+
+export const SearchInput = ({ onSearch = _.noop, onChange, ...props }) => {
+  const supportsSearch = 'onsearch' in document.documentElement
+
+  const searchEl = useRef()
+  const attachRef = node => {
+    if (node) {
+      const el = findDOMNode(node)
+      searchEl.current = el
+      el.addEventListener('search', onSearch)
+    } else {
+      searchEl.current.removeEventListener('search', onSearch)
+    }
+  }
+
+  return textInput(_.merge({
+    ref: supportsSearch ? attachRef : undefined,
+    type: 'search',
+    spellCheck: false,
+    style: { WebkitAppearance: 'none' },
+    onChange,
+    onKeyDown: supportsSearch ? undefined : e => { // to make firefox behave like webkit/blink
+      switch (e.which) {
+        case 13: // return
+          onSearch(e)
+          break
+        case 27: // escape
+          onChange(_.merge(e, { target: { value: '' } }))
+          setTimeout(() => onSearch(e), 0) // wait a tick to let the onChange fire first
```

Comment:
This seems fragile. Since the search event also includes the value, could we just call `onSearch` so the caller only sees one event at a time?